<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Pedro Lopez-Cabanillas &lt;plcl@users.sourceforge.net&gt;</author>
 <comment>
KMetronome - ALSA Sequencer based MIDI metronome
Copyright (C) 2005-2010 Pedro Lopez-Cabanillas &lt;plcl@users.sourceforge.net&gt;

This program is free software; you can redistribute it and/or modify 
it under the terms of the GNU General Public License as published by 
the Free Software Foundation; either version 2 of the License, or 
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, write to the Free Software
Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
MA 02110-1301, USA
</comment>
 <class>KmetronomeViewBase</class>
 <widget class="QWidget" name="KmetronomeViewBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>390</width>
    <height>307</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>32767</width>
    <height>32767</height>
   </size>
  </property>
  <property name="baseSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MIDI Metronome</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="0" column="0" rowspan="2">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="KPushButton" name="m_playbtn">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="whatsThis">
        <string>This button starts playing the rhythm pattern</string>
       </property>
       <property name="text">
        <string>Play</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="KPushButton" name="m_stopbtn">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="whatsThis">
        <string>This button stops playing the rhythm pattern</string>
       </property>
       <property name="text">
        <string>Stop</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="KPushButton" name="m_configbtn">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="whatsThis">
        <string>This button opens the configuration window</string>
       </property>
       <property name="text">
        <string>Config</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="KPushButton" name="m_exitbtn">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="whatsThis">
        <string>This button exits the program</string>
       </property>
       <property name="text">
        <string>Exit</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>28</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item row="0" column="1" colspan="2">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QLCDNumber" name="m_measureLCD">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>60</height>
        </size>
       </property>
       <property name="whatsThis">
        <string>Number of measures (bars) and beats played from beginning</string>
       </property>
       <property name="numDigits">
        <number>8</number>
       </property>
       <property name="mode">
        <enum>QLCDNumber::Dec</enum>
       </property>
       <property name="segmentStyle">
        <enum>QLCDNumber::Flat</enum>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="1" colspan="2">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <layout class="QFormLayout" name="formLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="textLabel3">
         <property name="text">
          <string>Beats/Bar:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="buddy">
          <cstring>m_beatsBar</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="KIntSpinBox" name="m_beatsBar">
         <property name="maximumSize">
          <size>
           <width>60</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="whatsThis">
          <string>This is the number of parts for each measure.&lt;br&gt;
Values are from 1 to 32.&lt;br&gt;
It is the numerator of the time signature specification.</string>
         </property>
         <property name="value">
          <number>4</number>
         </property>
         <property name="maxValue" stdset="0">
          <number>32</number>
         </property>
         <property name="minValue" stdset="0">
          <number>1</number>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="textLabel4">
         <property name="text">
          <string>Rhythm figure:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="buddy">
          <cstring>m_figure</cstring>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="KComboBox" name="m_figure">
         <property name="whatsThis">
          <string>This is the part figure for the rhythm.&lt;br&gt;
Values mean 1, 2, 4, 8, 16, 32 and 64.&lt;br&gt;
It is the denominator in the time signature specification.</string>
         </property>
         <property name="currentIndex">
          <number>0</number>
         </property>
         <property name="currentItem" stdset="0">
          <number>0</number>
         </property>
         <item>
          <property name="text">
           <string>Whole</string>
          </property>
          <property name="icon">
           <iconset resource="kmetronome.qrc">
            <normaloff>:/icons/1.png</normaloff>:/icons/1.png</iconset>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Half</string>
          </property>
          <property name="icon">
           <iconset resource="kmetronome.qrc">
            <normaloff>:/icons/2.png</normaloff>:/icons/2.png</iconset>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Quarter</string>
          </property>
          <property name="icon">
           <iconset resource="kmetronome.qrc">
            <normaloff>:/icons/4.png</normaloff>:/icons/4.png</iconset>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Eighth</string>
          </property>
          <property name="icon">
           <iconset resource="kmetronome.qrc">
            <normaloff>:/icons/8.png</normaloff>:/icons/8.png</iconset>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Sixteenth</string>
          </property>
          <property name="icon">
           <iconset resource="kmetronome.qrc">
            <normaloff>:/icons/16.png</normaloff>:/icons/16.png</iconset>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Thirty-Second</string>
          </property>
          <property name="icon">
           <iconset resource="kmetronome.qrc">
            <normaloff>:/icons/32.png</normaloff>:/icons/32.png</iconset>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Sixty-Fourth</string>
          </property>
          <property name="icon">
           <iconset resource="kmetronome.qrc">
            <normaloff>:/icons/64.png</normaloff>:/icons/64.png</iconset>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item row="2" column="0" colspan="2">
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="label_1">
       <property name="text">
        <string>Weak</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignBottom|Qt::AlignHCenter</set>
       </property>
       <property name="buddy">
        <cstring>m_dial1</cstring>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Strong</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignBottom|Qt::AlignHCenter</set>
       </property>
       <property name="buddy">
        <cstring>m_dial2</cstring>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Volume</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignBottom|Qt::AlignHCenter</set>
       </property>
       <property name="buddy">
        <cstring>m_dial3</cstring>
       </property>
      </widget>
     </item>
     <item row="0" column="3">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Pan</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignBottom|Qt::AlignHCenter</set>
       </property>
       <property name="buddy">
        <cstring>m_dial4</cstring>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="Knob" name="m_dial1">
       <property name="maximumSize">
        <size>
         <width>50</width>
         <height>50</height>
        </size>
       </property>
       <property name="whatsThis">
        <string>This is the velocity of the weak notes in the rhythm pattern (all of them, except the first one)</string>
       </property>
       <property name="maximum">
        <number>127</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="notchTarget">
        <double>10.000000000000000</double>
       </property>
       <property name="notchesVisible">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="Knob" name="m_dial2">
       <property name="maximumSize">
        <size>
         <width>50</width>
         <height>50</height>
        </size>
       </property>
       <property name="whatsThis">
        <string>This is the velocity of the strong note in the rhythm pattern (the first one)</string>
       </property>
       <property name="maximum">
        <number>127</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="notchTarget">
        <double>10.000000000000000</double>
       </property>
       <property name="notchesVisible">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="Knob" name="m_dial3">
       <property name="maximumSize">
        <size>
         <width>50</width>
         <height>50</height>
        </size>
       </property>
       <property name="whatsThis">
        <string>This is the MIDI volume control (CC7)</string>
       </property>
       <property name="maximum">
        <number>127</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="notchTarget">
        <double>10.000000000000000</double>
       </property>
       <property name="notchesVisible">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="3">
      <widget class="Knob" name="m_dial4">
       <property name="maximumSize">
        <size>
         <width>50</width>
         <height>50</height>
        </size>
       </property>
       <property name="whatsThis">
        <string>This is the MIDI pan control (CC10)</string>
       </property>
       <property name="maximum">
        <number>127</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="notchTarget">
        <double>10.000000000000000</double>
       </property>
       <property name="notchesVisible">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="2" column="2" rowspan="2">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="KComboBox" name="m_air">
       <property name="whatsThis">
        <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans Serif'; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;This is the &lt;span style=&quot; font-style:italic;&quot;&gt;aire&lt;/span&gt; control to change the tempo value using musical names&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLCDNumber" name="m_tempoLCD">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>150</width>
         <height>100</height>
        </size>
       </property>
       <property name="whatsThis">
        <string>The current &lt;i&gt;tempo&lt;/i&gt; value</string>
       </property>
       <property name="numDigits">
        <number>3</number>
       </property>
       <property name="value" stdset="0">
        <double>100.000000000000000</double>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="3" column="0" colspan="2">
    <layout class="QVBoxLayout">
     <item>
      <widget class="QLabel" name="textLabel1_2">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Tempo (in quarters per minute)</string>
       </property>
       <property name="wordWrap">
        <bool>false</bool>
       </property>
       <property name="buddy">
        <cstring>m_tempo</cstring>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="m_tempo">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="whatsThis">
        <string>This slider changes the &lt;i&gt;tempo&lt;/i&gt;, or speed of the rhythm.&lt;br&gt;
Values are 25 to 250 quarters per minute (&lt;i&gt;Maelzel Metronome units&lt;/i&gt;).</string>
       </property>
       <property name="minimum">
        <number>25</number>
       </property>
       <property name="maximum">
        <number>250</number>
       </property>
       <property name="value">
        <number>100</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::TicksBelow</enum>
       </property>
       <property name="tickInterval">
        <number>5</number>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>KIntSpinBox</class>
   <extends>QSpinBox</extends>
   <header>knuminput.h</header>
  </customwidget>
  <customwidget>
   <class>KPushButton</class>
   <extends>QPushButton</extends>
   <header>kpushbutton.h</header>
  </customwidget>
  <customwidget>
   <class>KComboBox</class>
   <extends>QComboBox</extends>
   <header>kcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>Knob</class>
   <extends>QDial</extends>
   <header>knob.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>m_playbtn</tabstop>
  <tabstop>m_stopbtn</tabstop>
  <tabstop>m_configbtn</tabstop>
  <tabstop>m_exitbtn</tabstop>
  <tabstop>m_beatsBar</tabstop>
  <tabstop>m_figure</tabstop>
  <tabstop>m_air</tabstop>
  <tabstop>m_dial1</tabstop>
  <tabstop>m_dial2</tabstop>
  <tabstop>m_dial3</tabstop>
  <tabstop>m_dial4</tabstop>
  <tabstop>m_tempo</tabstop>
 </tabstops>
 <includes>
  <include location="local">knob.h</include>
  <include location="local">knob.h</include>
  <include location="local">kpushbutton.h</include>
  <include location="local">kpushbutton.h</include>
  <include location="local">kpushbutton.h</include>
  <include location="local">kpushbutton.h</include>
  <include location="local">knuminput.h</include>
  <include location="local">kcombobox.h</include>
 </includes>
 <resources>
  <include location="kmetronome.qrc"/>
 </resources>
 <connections/>
</ui>
